kamon-statsd/src/main/scala/kamon/statsd/StatsDMetricsSender.scala

```diff
@@ -80,15 +80,15 @@ class StatsDMetricsSender(remote: InetSocketAddress, maxPacketSizeInBytes: Long,
 }
 
 object StatsDMetricsSender {
-  def props(remote: InetSocketAddress, maxPacketSize: Long, metricKeyGenerator: MetricKeyGenerator): Props =
-    Props(new StatsDMetricsSender(remote, maxPacketSize, metricKeyGenerator))
+  def props(statsDHost: String, statsDPort: Int, maxPacketSize: Long, metricKeyGenerator: MetricKeyGenerator): Props =
+    Props(new StatsDMetricsSender(statsDHost, statsDPort, maxPacketSize, metricKeyGenerator))
 }
 
 trait UdpExtensionProvider {
   def udpExtension(implicit system: ActorSystem): ActorRef = IO(Udp)
 }
 
-class MetricDataPacketBuilder(maxPacketSizeInBytes: Long, udpSender: ActorRef, remote: InetSocketAddress) {
+class MetricDataPacketBuilder(maxPacketSizeInBytes: Long, udpSender: ActorRef, statsDHost: String, statsDPort: Int) {
```

Comment:
I think there is no need to change the MetricDataPacketBuilder, it can receive `remote: InetSocketAddress` as it did before because typically all the UDP packets are dispatched within a few milliseconds and there is no chance that the cached address will change in that time given the minimum cache time is 1 second and the host is already resolved in line 51 (after the suggested change). Only one TickMetricSnapshot is received on every flush so, by default, this guy will only be created once every 10 seconds and live for a few milliseconds.

Another option would be to still receive the host and port and have a val with them inside MetricDataPacketBuilder, both approaches seem fine to me.